src/main/java/com/virtuslab/using_directives/custom/Scanner.java

```diff
@@ -203,22 +203,29 @@ public boolean isContinuing(Tokens lastToken) {
 
   private IndentWidth indentWidthRecur(
       int idx, char ch, int n, Function<IndentWidth, IndentWidth> k) {
-    if (idx < 0) return k.apply(new Run(ch, n));
-    else {
+    while (idx >= 0) {
       char nextChar = reader.buf[idx];
       if (nextChar == LF) return k.apply(new Run(ch, n));
       else if (nextChar == ' ' || nextChar == '\t') {
-        if (nextChar == ch) return indentWidthRecur(idx - 1, ch, n + 1, k);
-        else {
-          Function<IndentWidth, IndentWidth> k1;
-          if (n == 0) k1 = k;
-          else k1 = i -> new Conc(i, new Run(ch, n));
-          return indentWidthRecur(idx - 1, nextChar, 1, k1);
+        if (nextChar == ch) {
+          idx = idx - 1;
+          n = n + 1;
+        } else {
+          char c = ch;
+          int nn = n;
```

Comment:
`nn` is unnecessary

Reply:
Unfortunately, Java is very strict with local variables that it can take into closures.